src/simulation/simulation_state.jl

```diff
@@ -1,44 +1,107 @@
+        Dict{ConstraintKey, StateData}(),
+        Dict{AuxVarKey, StateData}(),
+        Dict{VariableKey, StateData}(),
     )
 end
 
 struct SimulationState
+    end_of_step_timestamp::Base.RefValue{Dates.DateTime}
     decision_states::StateInfo
     system_state::StateInfo
 end
 
+get_end_of_step_timestamp(s::SimulationState) = s.end_of_step_timestamp[]
+
+function set_end_of_step_timestamp!(s::SimulationState, val::Dates.DateTime)
+    s.end_of_step_timestamp[] = val
+    return
+end
+
 function SimulationState()
-    return SimulationState(StateInfo(), StateInfo())
+    return SimulationState(Ref(UNSET_INI_TIME), StateInfo(), StateInfo())
 end
 
 get_decision_states(s::SimulationState) = s.decision_states
 get_system_state(s::SimulationState) = s.system_state
 
+function _get_state_params(models::SimulationModels, simulation_step::Dates.Period)
+    params = OrderedDict{OptimizationContainerKey, NTuple{2, Dates.Millisecond}}()
+    for model in get_decision_models(models)
+        container = get_optimization_container(model)
+        model_resolution = get_resolution(model)
+        horizon_step = get_horizon(model) * model_resolution
+        for type in [:variables, :aux_variables]
+            field_containers = getfield(container, type)
+            for key in keys(field_containers)
+                if !haskey(params, key)
+                    params[key] = (max(simulation_step, horizon_step), model_resolution)
+                else
+                    params[key] = (
+                        max(params[key][1], horizon_step),
+                        min(params[key][2], model_resolution),
+                    )
+                end
+            end
+        end
+    end
+    return params
+end
+
 function _initialize_model_states!(
     states::StateInfo,
     model::OperationModel,
-    simulation_step::Dates.Period,
+    simulation_initial_time::Dates.DateTime,
+    params,
 )
     container = get_optimization_container(model)
-    model_resolution = get_resolution(model)
-    value_counts = simulation_step ÷ model_resolution
-    for type in [:variables, :aux_variables]
-        field_containers = getfield(container, type)
-        field_states = getfield(states, type)
+    for field in [:variables, :aux_variables]
+        field_containers = getfield(container, field)
+        field_states = getfield(states, field)
         for (key, value) in field_containers
             # TODO: Handle case of sparse_axis_array
-            column_names, _ = axes(value)
-            field_states[key] =
-                JuMP.Containers.DenseAxisArray{Float64}(undef, column_names, 1:value_counts)
+            value_counts = params[key][1] ÷ params[key][2]
+            if length(axes(value)) == 1
+                column_names = [string(encode_key(key))]
+            elseif length(axes(value)) == 2
+                column_names, _ = axes(value)
+            else
+                @warn("Multidimensional Array caching is not currently supported")
```

Comment:
Will everything still work in this case? Should it throw an error instead?

Reply:
I put a warning since at the moment isn't clear when this ill happen